src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+    fn parent_get_preferred_height_for_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        width: i32,
+    ) -> (i32, i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_preferred_height_for_width {
+                let mut minimum_size = mem::MaybeUninit::uninit();
+                let mut natural_size = mem::MaybeUninit::uninit();
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    width,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                );
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+
+    fn parent_get_aligned_area<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        flags: CellRendererState,
+        cell_area: &gdk::Rectangle,
+    ) -> gdk::Rectangle {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            let mut aligned_area = gdk::Rectangle::uninitialized();
+            if let Some(f) = (*parent_class).get_aligned_area {
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    flags.to_glib(),
+                    cell_area.to_glib_none().0,
+                    aligned_area.to_glib_none_mut().0,
+                );
+            }
+            aligned_area
+        }
+    }
+
+    fn parent_render<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        cr: &cairo::Context,
+        widget: &P,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).render {
```

Comment:
`gtk_cell_renderer_render()` does a critical warning but does not crash if it's `NULL`. So doing nothing as well as panicking should both be fine here. Your choice :)

Reply:
I'd rather not crash.

Reply:
`g_return_val_if_fail()` is basically like a panic in Rust, so that would be the consistent way with the C code. But just doing nothing here seems equally fine IMHO